XML parsing error in HawkSQLite-Q1 at size 512

An XML parsing exception occured during the latest measurements (commit 18d10187768af04f2249a0142b340c02dc91a3c9).

`output/details/output/log-2021-05-13T13.55.19.log:5660` in the paper repo:
```Running benchmark: tool = HawkSQLite, change set = 512, query = Q1
Exception in thread "main" org.eclipse.emf.ecore.resource.Resource$IOWrappedException: Element type "posts" must be followed by either attribute specifications, ">" or "/>".
	at org.eclipse.emf.ecore.xmi.impl.XMLLoadImpl.load(XMLLoadImpl.java:195)
	at org.eclipse.emf.ecore.xmi.impl.XMLResourceImpl.doLoad(XMLResourceImpl.java:261)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1563)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1342)
	at org.hawk.ttc2018.BatchLauncher.applyChanges(BatchLauncher.java:70)
	at org.hawk.ttc2018.AbstractLauncher.applyChanges(AbstractLauncher.java:221)
	at org.hawk.ttc2018.AbstractLauncher.run(AbstractLauncher.java:137)
	at org.hawk.ttc2018.BatchLauncher.main(BatchLauncher.java:105)
Caused by: org.xml.sax.SAXParseExceptionpublicId: file:/ttc/models/512/initial.xmi; systemId: file:/ttc/models/512/initial.xmi; lineNumber: 402621; columnNumber: 32; Element type "posts" must be followed by either attribute specifications, ">" or "/>".
	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
...
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.seekCloseOfStartTag(XMLDocumentFragmentScannerImpl.java:1394)
...
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:327)
	at org.eclipse.emf.ecore.xmi.impl.XMLLoadImpl.load(XMLLoadImpl.java:175)
	... 7 more
```

It is strange, because other XMI-based tools run successfully before (L5557):
```
Running benchmark: tool = HawkNeo4JIncUpdateQuery, change set = 512, query = Q1
```

The image used in this benchmark is available as [`ftsrg/ttc2018@sha256:e3cf270aa36f895c70459b39c8d7afa97b634ea315fe6855913be36a524411ef`](https://hub.docker.com/layers/ftsrg/ttc2018/java8/images/sha256-e3cf270aa36f895c70459b39c8d7afa97b634ea315fe6855913be36a524411ef).

I ran it again. This time there were no errors but it reached the time limit.
With an increased time limit, I could run it (only this tool, size, query) successfully 5 times.

For comparison, the previous measurement also reached timeout for this:
`output/details/output-round-10/log-2021-05-09T18.30.46.log:5524` (commit cd5c9686f4810d425a33271d98c915dfb201ec2a):
```
Running benchmark: tool = HawkSQLite, change set = 512, query = Q1
Program reached the timeout set (600 s): tool = HawkSQLite, change set = 512, query = Q1, command: 'mvn -f ../Hawk/org.hawk.ttc2018/pom.xml -quiet -Pbatch -Dbackend=sqlite exec:exec'
```

I would suspect that a previous tool damaged the XMI file and the [restoration](https://github.com/TransformationToolContest/ttc2018liveContest/blob/ebb9bedc98d2c91842730a650aa8048d99df69c0/scripts/run.py#L103) was not complete, but the last run of that size finished on time.

--- a/solutions/Hawk/org.hawk.ttc2018/src/org/hawk/ttc2018/AbstractLauncher.java
+++ b/solutions/Hawk/org.hawk.ttc2018/src/org/hawk/ttc2018/AbstractLauncher.java
@@ -32,6 +32,7 @@ public abstract class AbstractLauncher {
 
 	private static final Logger LOGGER = LoggerFactory.getLogger(AbstractLauncher.class);
 	public static final String INITIAL_MODEL_FILENAME = "initial.xmi";
+	public static final String BACKUP_MODEL_FILENAME = "initial.xmi.backup";
 
 	protected final LauncherOptions opts;
 	private final List<Snapshot> results = new ArrayList<>();
@@ -119,8 +120,14 @@ public abstract class AbstractLauncher {
 
 		// Create a backup copy of the initial XMI
 		final File fInitial = new File(opts.getChangePath(), INITIAL_MODEL_FILENAME);
-		final File tmpBackup = File.createTempFile("backupInitial", ".xmi");
-		Files.copy(fInitial, tmpBackup);
+		final File fBackup = new File(opts.getChangePath(), BACKUP_MODEL_FILENAME);
+		if (!fBackup.exists()) {
+			// First run with this initial.xmi: back it up before working on it
+			Files.copy(fInitial, fBackup);
+		} else {
+			// Later run with this initial.xmi: restore the backup
+			Files.copy(fBackup, fInitial);
+		}
 
 		hawk = createHawk();
 		try {
@@ -140,20 +147,6 @@ public abstract class AbstractLauncher {
 
 		} finally {
 			hawk.shutdown();
-
-			/*
-			 * Reset the input model after we are done (originally at the beginning,
-			 * change requested by Georg to avoid interfering with other solutions.
-			 *
-			 * Use a plain copy instead of depending on 'git checkout', since the Docker
-			 * images do not use a git clone.
-			 */
-			try {
-				Files.copy(tmpBackup, fInitial);
-				tmpBackup.delete();
-			} catch (IOException e) {
-				LOGGER.error(e.getMessage(), e);
-			}
 		}
 	}
 

